src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -577,6 +580,39 @@ public boolean hasTag(String tag) {
 				log.printError();
 				return null;
 			}
+			Matcher literalClassInfoMatcher = LITERAL_CLASSINFO_PARSE.matcher(expr);
+			if (literalClassInfoMatcher.matches()) {
```

Comment:
since this is going to be called so often, I think it'd be great to check if the input string has an opening parentheses and ends with an closing parentheses before we try to match it to an expensive regex